tests/handlers/test_dashboards.py

```diff
@@ -76,13 +86,13 @@ def test_get_dashboard_filters_unauthorized_widgets(self):
         dashboard.layout = "[[{}, {}]]".format(widget.id, restricted_widget.id)
         db.session.commit()
 
-        rv = self.make_request("get", "/api/dashboards/{0}".format(dashboard.slug))
+        rv = self.make_request("get", "/api/dashboards/{0}".format(dashboard.id))
         self.assertEqual(rv.status_code, 200)
         self.assertTrue(rv.json["widgets"][0]["restricted"])
         self.assertNotIn("restricted", rv.json["widgets"][1])
 
     def test_get_non_existing_dashboard(self):
-        rv = self.make_request("get", "/api/dashboards/not_existing")
+        rv = self.make_request("get", "/api/dashboards/-1")
```

Comment:
This actually doesn't happen only to the dashboard, but anything that's `/api/resource/not_a_number` causes a sqlalchemy error. This PR didn't feel like a scope to fix it and at least now I don't have an idea on how to solve this now as it turned to be a dynamic thing.